kb(carousel):thumbnail scrollable navigation (#743)
telerik/blazor-docs: knowledge-base/carousel-thumbnail-scrollable-navigation.md

---

> title: Carousel Thumbnail Scrollable navigation

Reviewer: ```suggestion
title: Carousel Thumbnail Scrollable Navigation
```

> description: How to create Thumbnail Scrollable Navigation in Carousel?

Reviewer: ```suggestion
description: How to use thumbnails in scrollable navigation for the Carousel?
```

type: how-to

> page_title: Carousel Thumbnail Scrollable navigation

Reviewer: ```suggestion
page_title: Carousel Thumbnail Scrollable Navigation
```

slug: carousel-kb-thumbnail-scrollable-navigation
position: 
tags: telerik,blazor,carousel,thumbnail,navigation,images
ticketid: 1550292
res_type: kb
---

## Environment
<table>
	<tbody>
		<tr>
			<td>Product</td>
			<td>Carousel for Blazor</td>
		</tr>
	</tbody>
</table>

## Description
How to add a thumbnail scrollable navigation below the page/image in Carousel? This would be a nice addition, which would allow independent scrolling from that of the current page. It will also be clickable in the same way that the dots are to jump to the selected page.

## Solution
To add a thumbnail scrollable navigation:

> 1. Add HTML `<div>` container under the `Carousel` markup.
> 2. Apply custom CSS with the needed styles to the container.
> 3. Inside the `<div>`, loop through all the images in `Carousel` and define them in a smaller size.
> 4. Optionally, use a javascript function that keeps the synchronization of the `AutomaticPageChange` and the thumbnail.

Reviewer: ```suggestion
1. Render the thumbnails in a `<div>` container under the `Carousel`.
1. Use a `click` handler for the thumbnails to page the Carousel programmatically.
1. Optionally, use JavaScript to synchronize the scroll position of the thumbnails with the current Carousel page. This applies to the case when the user pages the Carousel with the built-in component buttons.
```

![](images/carousel-thumbnail-navigation.png)

>caption Component

````CSHTML
@inject IJSRuntime JSRuntime;

<TelerikCarousel Data="@CarouselData"
                 Width="600px" Height="384px"
                 PageChanged="PageChangedHandler" Page="PageIndex">
    <Template>
        <div class="image-with-text">
            <p>Showing image @(context.ImageID) of @CarouselData.Count.ToString().</p>
            <img src="@context.ImageUrl" alt="Photograph" width="612" height="384" />
        </div>
    </Template>
</TelerikCarousel>

<div class="container-nav">
    <div class="images-nav">
        @foreach (var img in CarouselData)
        {
            <img @onclick="@(() => PageChangedHandler(img.ImageID))" id="@img.ImageID"
                 class="image-thumbnail"
                 src="@img.ImageUrl" alt="Photograph" />
        }
    </div>
</div>

> <script suppress-error="BL9992">

Reviewer: ```suggestion
<!-- Move the JavaScript code to a separate .js file in the actual app -->
<script suppress-error="BL9992">
```

    function ScrollToCurrentPage(imgId) {
        var elem = document.getElementById(imgId);

>         console.log(imgId);
>         console.log(elem);

Reviewer: ```suggestion
```
